### ISSUE TITLE
[Suggestion] Minetweaker Support

### ISSUE THREAD

**Issue by WolfieWaffle** [einsteinsci/betterbeginnings#88]
TL;DR: Please add support for Minetweaker in the 1.7.10 version of this mod.

I really like the look of this mod, and especially the advanced crafting table. I would really like to make a modpack based off it, and I could do textures for the mod too (If my profile pic is still a gold-bordered book, I made that, for reference). Anyways I'm getting off track. I would really love to have support for Minetweaker, which if you aren't familiar with it it can modify recipes for vanilla and a lot of other mods. It has become really popular due to it's versatility and the amount of mods it supports. If you would add support for it, that would be amazing. I think it has an API, and it's main page is here: http://minetweaker3.powerofbytes.com/

Specifically, I think you should add support for the double crafting table (obviously) and the different furnaces. It would really be great for this to be added in the 1.7.10 version, because that's the version Minetweaker is currently in, as well as almost every mod used in modpacks.

Anyways, I think that Minetweaker support would make this mod very versatile and modpack makers (like me!) would be able to add other mod recipes to the mod. It already fits well with Iguana Tweaks and Hunger Overhaul, but mods like Bibliocraft or Balkon's Weapons still use the Vanilla table, and could be interesting with this mods crafting system. I hope you consider it! This mod looks promising, like Terrafirmacraft but a LOT lighter and less game changing. The only thing it needs to fit with other mods is the ability to use the new crafting table.

**Comment by einsteinsci** [comment]
I already have a feature just like this planned for the 1.0 release to be built-in to the mod, without any dependency on Minetweaker. To put it another way, in the 1.0 release, you will be able to add/change/remove advanced crafting recipes and special furnace recipes from a specific file in the BetterBeginnings config folder. This will be the main feature for 1.0.

However, I do not have any intention of pushing 1.7.10 past 0.9.5 besides bugfixes, and here's why:
* Minecraft 1.9 is rapidly approaching. As has become fairly standard in the modding scene, support for mods that are two versions old is *very quickly* dropped. I don't want to split my focus across three versions of my mod when I already have quite limited time as it is. I have to wrap up the 1.7.10 version quickly, and adding more features to it only slows it down.
* 1.7.10 mods are already in decline. Most of the mods that still have updates, excluding a few very large mods, have updated to 1.8, mine included. The way I code things for 1.7.10 is by making all my additions, tweaks, and bugfixes in the 1.8 version first, and then copy all the relevant stuff over to 1.7.10 when I know the new code works. 1.8 is the main and more popular version of BetterBeginnings, which means that to be able to keep up my development cycle (and all the other planned features), I'll need to single a specific version; in this case 1.8.
* 1.8 will likely be short-lived in the mod community, as 1.9 contains few structural changes besides dual-wielding and attack cooldown, and thus mod updates will be fairly quick. Compare this to 1.8, which is mostly back-end structural changes, where modders are still updating almost a year into the update. You'll see that it results in very little time for 1.8 modpacks. With the rush to create 1.8 modpacks before Forge reaches 1.9, 1.7.10 modpacks will dissipate very quickly.
* This feature I have planned is quite complex, so it will take quite some time to code in. Since 1.7.10 is already declining, it is quite likely that by the time I finish coding this for 1.8, there will be little emphasis on 1.7.10 modpacks, Minetweaker will be updated to 1.8, and this feature will no longer be needed for 1.7.10.

Furthermore, I have good reason for integrating this kind of feature directly into my mod with its own config, rather than offloading it to Minetweaker. Most modpack creators that will use my mod will have the same need to add recipes you described above for any or all other content mods they add to the modpack. However, not all will have a need to tweak the recipes using vanilla or other mods' methods (crafting, pulverizing, etc.), so there is greater need for the addition/customization of BetterBeginnings recipes than there is a need for Minetweaker in general. Thus, I will have this feature built in to my mod with no dependency.

**Comment by WolfieWaffle** [comment]
That sounds good. I guess I didn't realize that 1.7.10 was declining because I haven't seen many 1.8 mods and 1.9 is not out yet. I guess it's kind of in a limbo between updates, with half of the mods being updated and half not. But the reasons you gave make sense.
